src/cfile/core.py

```diff
@@ -136,7 +136,7 @@ class DataType(Element):
     """
     Base class for all data types
     """
-    def __init__(self, name: str | None) -> None:
+    def __init__(self, name: str = "") -> None:
```

Comment:
Less complicated API. Because what is difference between "" or None? You are checking for None but not for "".  So it is simpler and easier to check for "" then for None and ""

Reply:
So based on what you said bellow i understand why you are using None, but still i think it is redundant.